source/Phrase.cpp

```diff
@@ -13,11 +13,31 @@ PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 #include "Phrase.h"
 
 #include "DataNode.h"
+#include "Format.h"
 #include "GameData.h"
 #include "Random.h"
 
 using namespace std;
 
+namespace {
+	// Replace oldStr with newStr in target.
+	string Replace(const string &target, const string &oldStr, const string &newStr)
```

Comment:
Should this be here, or in `Format`?

Also, you use a different algorithm for this method and for the your new Format::Replace method - any particular reason?

Reply:
This function implements "replace" child node like  s///g including no regex, and treats the bracket ${ } as normal characters.
I would like to place here.